documents/model.go

```diff
@@ -740,17 +749,197 @@ func (m *CoreDocumentModel) ValidateDocumentAccess(docReq *p2ppb.GetDocumentRequ
+	}
+	nft.TokenId = tokenID
+
+	if grantReadAccess {
+		err = ndm.AddNFTToReadRules(registry, tokenID)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return ndm, ndm.setCoreDocumentSalts()
+}
+
+// IsNFTMinted checks if the there is an NFT that is minted against this document in the given registry.
+func (m *CoreDocumentModel) IsNFTMinted(tokenRegistry TokenRegistry, registry common.Address) bool {
+	nft := getStoredNFT(m.Document.Nfts, registry.Bytes())
+	if nft == nil {
+		return false
+	}
+
+	_, err := tokenRegistry.OwnerOf(registry, nft.TokenId)
+	return err == nil
+}
+
+func getStoredNFT(nfts []*coredocumentpb.NFT, registry []byte) *coredocumentpb.NFT {
+	for _, nft := range nfts {
+		if bytes.Equal(nft.RegistryId[:20], registry) {
+			return nft
+		}
+	}
+
+	return nil
+}
+
+// IsAccountInRole checks if the the account exists in the role provided
+func (m *CoreDocumentModel) IsAccountInRole(roleKey []byte, account identity.CentID) bool {
+	role, err := getRole(roleKey, m.Document.Roles)
+	if err != nil {
+		return false
+	}
+
+	_, found := isAccountInRole(role, account)
+	return found
+}
+
+// GetNFTProofs generate proofs required to mint an NFT
+func (m *CoreDocumentModel) GetNFTProofs(
+	dataRoot []byte,
+	account identity.CentID,
+	registry common.Address,
+	tokenID, roleProof []byte,
+	nftUniqueProof, readAccessProof bool) (proofs []*proofspb.Proof, err error) {
+
+	var pfKeys []string
+	if roleProof != nil {
+		pk, err := getRoleProofKey(m.Document.Roles, roleProof, account)
+		if err != nil {
+			return nil, err
+		}
+
+		pfKeys = append(pfKeys, pk)
+	}
+
+	if nftUniqueProof {
+		pk, err := getNFTUniqueProofKey(m.Document.Nfts, registry)
+		if err != nil {
+			return nil, err
+		}
+
+		pfKeys = append(pfKeys, pk)
+	}
+
+	if readAccessProof {
+		pks, err := getReadAccessProofKeys(m, registry, tokenID)
+		if err != nil {
+			return nil, err
+		}
+
+		pfKeys = append(pfKeys, pks...)
+	}
+
+	signingRootProofHashes, err := m.getSigningRootProofHashes()
+	if err != nil {
+		return nil, errors.New("failed to generate signing root proofs: %v", err)
+	}
+
+	cdtree, err := m.GetDocumentTree()
+	if err != nil {
+		return nil, errors.New("failed to generate document tree: %v", err)
+	}
+
+	// We support fields that belong to different document trees, as we do not prepend a tree prefix to the field, the approach
+	// is to try in both trees to find the field and create the proof accordingly
+	for _, field := range pfKeys {
+		proof, err := cdtree.CreateProof(field)
+		if err != nil {
+			return nil, errors.New("failed to create proof: %v", err)
+		}
+
+		proof.SortedHashes = append(proof.SortedHashes, dataRoot)
+		proof.SortedHashes = append(proof.SortedHashes, signingRootProofHashes...)
+		proofs = append(proofs, &proof)
+	}
+
+	return proofs, nil
+}
+
+func getReadAccessProofKeys(m *CoreDocumentModel, registry common.Address, tokenID []byte) (pks []string, err error) {
+	var rridx int  // index of the read rules which contain the role
```

Comment:
Abbreviations need to be capitalised right? eg: rridx -> RRIDX

Reply:
not when they are starting with abbrevations like `id`

Reply:
Example:
```
tokenID
id
idSomething
```

Reply:
I don't get it, in this case the entire variable name is an abbreviation right? not just the start.

Reply:
I agree with ved, internal abbreviated short variables like this are better than rRIDX or something similar.

Reply:
@lucasvo it wont be like that, according to go convention the variable name should be RRIDX

Reply:
@vimukthi-git it should be small abbreviation if the variable name contains only that example: `id`.
This would change if the variable name contains more than an abbreviation. Example "tokenID". 
If the variable name starts with the abbreviation it would be like this `xmlHTTPRequest`.

More here - https://github.com/golang/go/wiki/CodeReviewComments#initialisms